src/main/java/ee/carlrobert/codegpt/completions/llama/LlamaServerAgent.java

```diff
@@ -32,65 +35,94 @@ public final class LlamaServerAgent implements Disposable {
 
   private @Nullable OSProcessHandler makeProcessHandler;
   private @Nullable OSProcessHandler startServerProcessHandler;
+  private ServerProgressPanel activeServerProgressPanel;
+  private boolean stoppedByUser;
 
   public void startAgent(
       LlamaServerStartupParams params,
       ServerProgressPanel serverProgressPanel,
       Runnable onSuccess,
-      Runnable onServerTerminated) {
+      Consumer<ServerProgressPanel> onServerTerminated) {
```

Comment:
Changed to a `Consumer<ServerProgressPanel>` because if you:
1. Open CodeGPT Settings and start the server
2. Close the Settings
3. Reopen the Settings
4. Stop the Server, the text in the `ServerProgressPanel` is not correctly set to "Terminated Server" because an old reference to the `ServerProgressPanel` is used, therefore I changed it to always use the correct `ServerProgressPanel`instance.